sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java

```diff
@@ -208,32 +294,157 @@ public ContactDirectorySteps(
         });
     And(
         "I click on All button in Contact Directory Page",
         () -> {
-          webDriverHelpers.clickOnWebElementBySelector(ALLBUTTON_CONTACT);
-          TimeUnit.SECONDS.sleep(8); // needed for table refresh
+          webDriverHelpers.waitUntilAListOfWebElementsAreNotEmpty(ALLBUTTON_CONTACT);
+          webDriverHelpers.waitUntilAListOfWebElementsAreNotEmpty(ALLBUTTON_CONTACT);
         });
     And(
         "I click on Converted to case pending button on Contact Directory Page",
         () -> {
-          webDriverHelpers.clickOnWebElementBySelector(CONVERTED_TO_CASE_BUTTON);
-          TimeUnit.SECONDS.sleep(8); // needed for table refresh
+          webDriverHelpers.waitUntilAListOfWebElementsAreNotEmpty(CONVERTED_TO_CASE_BUTTON);
+          webDriverHelpers.waitUntilAListOfWebElementsAreNotEmpty(CONVERTED_TO_CASE_BUTTON);
         });
     And(
         "I click on Active contact button in Contact Directory Page",
         () -> {
-          webDriverHelpers.clickOnWebElementBySelector(ACTIVE_CONTACT_BUTTON);
-          TimeUnit.SECONDS.sleep(8); // needed for table refresh
+          webDriverHelpers.waitUntilAListOfWebElementsAreNotEmpty(ACTIVE_CONTACT_BUTTON);
+          webDriverHelpers.waitUntilAListOfWebElementsAreNotEmpty(ACTIVE_CONTACT_BUTTON);
         });
     And(
         "I click on Dropped button on Contact Directory Page",
         () -> {
-          webDriverHelpers.clickOnWebElementBySelector(DROPPED_BUTTON);
-          TimeUnit.SECONDS.sleep(8); // needed for table refresh
+          webDriverHelpers.waitUntilAListOfWebElementsAreNotEmpty(DROPPED_BUTTON);
+          webDriverHelpers.waitUntilAListOfWebElementsAreNotEmpty(DROPPED_BUTTON);
+        });
+
+    When(
+        "I select all options in Type of activity from Combobox in Exposure form",
+        () -> {
+          selectAllActivityTypes();
+        });
+
+    When(
+        "I select ([^\"]*) option in Type of activity from Combobox in Exposure form",
+        (String typeOfactivity) -> {
+          webDriverHelpers.selectFromCombobox(TYPE_OF_ACTIVITY_COMBOBOX, typeOfactivity);
+        });
+
+    When(
+        "I select all Type of gathering from Combobox in Exposure form",
+        () -> {
+          selectAllGatheringType();
+        });
+
+    When(
+        "I check all Type of place from Combobox in Exposure form",
+        () -> {
+          selectAllTypeOfPlace();
+        });
+
+    When(
+        "I fill Location form for Type of place by options excluded Other and Facility",
+        () -> {
+          webDriverHelpers.selectFromCombobox(TYPE_OF_PLACE_COMBOBOX, TypeOfPlace.HOME.getPlace());
+          exposureData = contactService.buildGeneratedExposureDataContactForRandomInputs();
+          fillLocation(exposureData);
+          webDriverHelpers.clickOnWebElementBySelector(DONE_BUTTON);
+        });
+
+    When(
+        "I fill Location form for Type of place by Other option",
```

Comment:
this method actually generates some random data and fills 1 field.
I don't see the need of building a new DTO only to update one field value.